### PR TITLE
Add configuration options for spiflash

### PR DESCRIPTION
**Pull request by moidx**
* Make spiflash configuratble from capture.yaml.
* Add FTDI connectivity instructions to the doc/getting_started.md
  guide.

Signed-off-by: Miguel Osorio <miguelosorio@google.com>

### CODE REVIEW

File: cw/cw305/util/spiflash.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+r"""Utilities used to download firmware images into the OpenTitan system."""
```
Comment (alphan) [inline cw/cw305/util/spiflash.py:4]: I don't think this should be a raw string.

File: cw/cw305/util/spiflash.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  See doc/getting_started.md for details on how to connect the FTDI cable to
+  the CW305 FPGA target board.
+  """
+  def __init__(self, spiflash_path, dev_id, dev_sn, input):
+    """Initializes spiflash FTDI programmer.
+
+    @param spiflash_path Path to spiflash executable.
+    @param dev_id FTDI USB device ID in vendor:product hex format.
+    @param dev_sn FTDI USB serial number as reported by lsusb.
+    @param input Path to OpenTitan firmware image.
```
Comment (alphan) [inline cw/cw305/util/spiflash.py:20]: Are we using doxygen for python as well? If not, we should probably follow [this](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods).

File: cw/cw305/capture_traces.py [hunk 3]
```diff
@@ -14,13 +14,20 @@
 
 from util import device
 from util import plot
+from util import spiflash
 
 
-def initialize_test(device_cfg):
+def initialize_test(device_cfg, spiflash_cfg):
```
Comment (alphan) [inline cw/cw305/capture_traces.py:20]: Maybe this should be called `initialize_capture` for the sake of clarity (+ the docstring below).

File: cw/cw305/util/spiflash.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    """
+    self.cmd = [spiflash_path, f'--dev-id={dev_id}', f'--dev-sn={dev_sn}',
+                f'--input={input}']
+
+  def run(self):
+    """Run spiflash utility with predifined command line arguments."""
```
Comment (alphan) [inline cw/cw305/util/spiflash.py:26]: typo: predifined

File: doc/getting_started.md [hunk 5]
```diff
@@ -7,6 +7,26 @@ In the meantime, check the following PRs:
 * https://github.com/lowRISC/opentitan/pull/2587
 * https://github.com/lowRISC/opentitan/pull/2735
 
+### Downloading code via FTDI
+
+Currently the target firmware is downloaded using the
+[spiflash](https://docs.opentitan.org/sw/host/spiflash/README/) utility. A
+debian-compatible pre-built binary is available at cw/cw305/bin/linux/spiflash.
```
Comment (vogelpi) [inline doc/getting_started.md:14]: I understand that it is convenient to "ship" this and other binaries (bitstreams) directly with the code in the same repo. I also think this is okay now that we are at the very beginning as it eases collaboration and allows to get up to speed more easily.

However, I don't think we should keep updating these binaries. Some of them are pretty big and they will remain in the tree forever. Once we have things up and running, I suggest we kick the binaries out again and create a script to download them from GitHub releases. There we can periodically provide new binaries as we do with the toolchains https://github.com/lowRISC/lowrisc-toolchains/releases.

Comment (moidx) [reply doc/getting_started.md:14]: SGTM. That's a good idea :)

File: cw/cw305/util/spiflash.py (hunk 1 above)
Comment (moidx) [reply cw/cw305/util/spiflash.py]: Done.

File: cw/cw305/util/spiflash.py (hunk 2 above)
Comment (moidx) [reply cw/cw305/util/spiflash.py]: SGTM. I prefer the format you suggested.

I will let lowRISC weigh on this for additional coding style guidelines across projects.

File: cw/cw305/capture_traces.py (hunk 3 above)
Comment (moidx) [reply cw/cw305/capture_traces.py]: Done.

File: cw/cw305/util/spiflash.py (hunk 4 above)
Comment (moidx) [reply cw/cw305/util/spiflash.py]: Fixed.

**Pull request merged**